live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/UrlStorageClientTest.java

```diff
@@ -0,0 +1,134 @@
+    UrlStorageClient urlStorageClient;
+
+    @Captor
+    ArgumentCaptor<StorageLocationRequest> storageLocationRequestCaptor;
+    @Captor
+    ArgumentCaptor<FileUploadRequest> fileUploadRequestCaptor;
+
+    @Test
+    void testUploadDataFromFile() throws IOException
+    {
+        // given
+        URL url = mock(URL.class);
+        File file = Files.createTempFile("test", ".txt").toFile();
+        Files.write(file.toPath(), FILE_CONTENT.getBytes(), APPEND);
+        given(storageLocationRequesterMock.requestStorageLocation(any())).willReturn(url);
+
+        // when
+        urlStorageClient.upload(file, FILE_CONTENT_TYPE, NODE_ID);
+
+        // then
+        then(storageLocationRequesterMock).should().requestStorageLocation(storageLocationRequestCaptor.capture());
+        StorageLocationRequest actualStorageLocationRequest = storageLocationRequestCaptor.getValue();
+        assertThat(actualStorageLocationRequest).satisfies(request -> {
+            assertThat(request.nodeId()).isEqualTo(NODE_ID);
+            assertThat(request.contentType()).isEqualTo(FILE_CONTENT_TYPE);
+        });
+        then(fileUploaderMock).should().upload(fileUploadRequestCaptor.capture());
+        FileUploadRequest actualFileUploadRequest = fileUploadRequestCaptor.getValue();
+        assertThat(actualFileUploadRequest).satisfies(request -> {
+            assertThat(request.contentType()).isEqualTo(FILE_CONTENT_TYPE);
+            assertThat(request.storageLocation()).isEqualTo(url);
+            assertThatExceptionOfType(ClosedChannelException.class).isThrownBy(() -> request.inputStream().read());
+        });
+    }
+
+    @Test
+    void testUploadDataFromInputStream() throws IOException
+    {
+        // given
+        URL url = mock(URL.class);
+        @Cleanup
+        InputStream inputStream = new ByteArrayInputStream(FILE_CONTENT.getBytes());
+        given(storageLocationRequesterMock.requestStorageLocation(any())).willReturn(url);
+
+        // when
+        urlStorageClient.upload(inputStream, FILE_CONTENT_TYPE, NODE_ID);
+
+        // then
+        then(storageLocationRequesterMock).should().requestStorageLocation(storageLocationRequestCaptor.capture());
+        StorageLocationRequest actualStorageLocationRequest = storageLocationRequestCaptor.getValue();
+        assertThat(actualStorageLocationRequest).satisfies(request -> {
+            assertThat(request.nodeId()).isEqualTo(NODE_ID);
+            assertThat(request.contentType()).isEqualTo(FILE_CONTENT_TYPE);
+        });
+        then(fileUploaderMock).should().upload(fileUploadRequestCaptor.capture());
+        FileUploadRequest actualFileUploadRequest = fileUploadRequestCaptor.getValue();
+        assertThat(actualFileUploadRequest).satisfies(request -> {
+            assertThat(request.contentType()).isEqualTo(FILE_CONTENT_TYPE);
+            assertThat(request.storageLocation()).isEqualTo(url);
+            assertThat(request.inputStream()).hasContent(FILE_CONTENT);
+        });
```

Comment:
I would change both tests to look more like this as it will be a lot more readable:
```
    @Test
    void testUploadDataFromInputStream()
    {
        // given
        InputStream testData = mock();
        
        URL testUrl = mock();
        given(storageLocationRequesterMock.requestStorageLocation(any())).willReturn(testUrl);

        // when
        urlStorageClient.upload(testData, FILE_CONTENT_TYPE, NODE_ID);

        // then
        StorageLocationRequest expectedStorageLocationRequest = new StorageLocationRequest(NODE_ID, FILE_CONTENT_TYPE);
        then(storageLocationRequesterMock).should().requestStorageLocation(eq(expectedStorageLocationRequest));

        FileUploadRequest expectedFileUploadRequest = new FileUploadRequest(testData, FILE_CONTENT_TYPE, testUrl);
        then(fileUploaderMock).should().upload(expectedFileUploadRequest);
    }
```

Reply:
I'll check it, I'm not sure if it will pass the equality check on `InputStream` 🤔

Reply:
Works fine, thank for the hint! Fixed